roofit/roofitcore/src/RooGradMinimizerFcn.cxx

```diff
@@ -64,7 +64,7 @@ ROOT::Math::IMultiGradFunction *RooGradMinimizerFcn::Clone() const
 void RooGradMinimizerFcn::synchronizeGradientParameterSettings(
    std::vector<ROOT::Fit::ParameterSettings> &parameter_settings) const
 {
-   _gradf.SetInitialGradient(_context->getMultiGenFcn(), parameter_settings, _grad);
+   _gradf.SetInitialGradient(parameter_settings, _grad);
```

Comment:
```suggestion
   _gradf.SetInitialGradient(nullptr, parameter_settings, _grad);
```
There is also another solution to the problem of the dynamic cast in the constructor: you can just pass a `nullptr` as the first argument, which is unused anyway.

I would feel more comfortable with that, because it doesn't require changing the public interface of `NumericalDerivator::SetInitialGradient`.